### PR TITLE
check description + code review

### PR DESCRIPTION
**Pull request by EquipableMC**
improves syntax, changes list so it's done through essentials and be less complex, added a way to list all the players in god mode, and removes unneeded imports

### CODE REVIEW

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java [hunk 1]
```diff
@@ -39,12 +36,9 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean kleenean
...
         List<Player> vanishedPlayers = new ArrayList<>();
-        for (Player player : Bukkit.getOnlinePlayers()) {
-            User user = SkEssentials.essentials.getUser(player);
-            if (user != null) {
-                if (user.isVanished()) {
-                    vanishedPlayers.add(player);
-                }
+        for (User user : SkEssentials.essentials.getOnlineUsers()) {
+            if (user.isVanished()) {
+                vanishedPlayers.add(user.getBase());
```
Comment (cheeezburga) [inline src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java:41]: See #3 for this I reckon

Comment (EquipableMC) [reply src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java:41]: ah shoot

Comment (EquipableMC) [reply src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java:41]: wait what do you mean

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprGodModePlayers.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Override
+    public Class<? extends Player> getReturnType() {
+        return Player.class;
+    }
+
+    @Override
+    public String toString(@Nullable Event event, boolean b) {
+        return "all gode mode players";
```
Comment (Sparky200) [inline src/main/java/site/equipable/skEssentials/skript/expressions/ExprGodModePlayers.java:59]:
```suggestion
        return "all god mode players";
```

**Review by Sparky200 — APPROVED** [review]

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprAfkPlayers.java [hunk 3]
```diff
@@ -22,14 +22,14 @@
...
-@Name("Afk Players")
+@Name("All Afk Players")
 @Description("The list of afk players on the server.")
 @Examples({"send \"%all of the afk players%\""})
 @Since("1.0.0")
 public class ExprAfkPlayers extends SimpleExpression<Player> {
 
     static {
-        Skript.registerExpression(ExprAfkPlayers.class, Player.class, ExpressionType.SIMPLE, "[all [[of] the]|the] (afk|away from keyboard|idle) players");
+        Skript.registerExpression(ExprAfkPlayers.class, Player.class, ExpressionType.SIMPLE, "[all] [[of] the] (afk|away from keyboard|idle) players");
```
Comment (sovdeeth) [inline src/main/java/site/equipable/skEssentials/skript/expressions/ExprAfkPlayers.java:32]: lime led you astray with this change
the previous one was better

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprGodModePlayers.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.List;
+
+@Name("All God Mode Players")
+@Description("The list of players in god mode on the server.")
+@Examples({"send \"%all of the god mode players%\""})
+@Since("1.0.0")
+public class ExprGodModePlayers extends SimpleExpression<Player> {
+
+    static {
+        Skript.registerExpression(ExprGodModePlayers.class, Player.class, ExpressionType.SIMPLE, "[all] [[of] the] god mode players");
```
Comment (sovdeeth) [inline src/main/java/site/equipable/skEssentials/skript/expressions/ExprGodModePlayers.java:28]: same here, but also `players (in|with) god mode` sounds better to me

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java [hunk 5]
```diff
@@ -28,7 +25,7 @@
 public class ExprVanishedPlayers extends SimpleExpression<Player> {
 
     static {
-        Skript.registerExpression(ExprVanishedPlayers.class, Player.class, ExpressionType.SIMPLE, "[all [[of] the]|the] vanish[ed] players");
```
Comment (sovdeeth) [inline src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java:31]: same as before

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprGodModePlayers.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.List;
+
+@Name("All God Mode Players")
+@Description("The list of players in god mode on the server.")
+@Examples({"send \"%all of the god mode players%\""})
+@Since("1.0.0")
+public class ExprGodModePlayers extends SimpleExpression<Player> {
+
+    static {
+        Skript.registerExpression(ExprGodModePlayers.class, Player.class, ExpressionType.SIMPLE, "players (in|with) god mode");
```
Comment (sovdeeth) [inline src/main/java/site/equipable/skEssentials/skript/expressions/ExprGodModePlayers.java:28]: where'd `all the` go

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprGodModePlayers.java (hunk 6 above)
Comment (EquipableMC) [reply src/main/java/site/equipable/skEssentials/skript/expressions/ExprGodModePlayers.java:28]: I'm adding it back, I am about to commit (I was literally about to click the commit button 😂)

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprAfkPlayers.java [hunk 7]
```diff
@@ -22,14 +22,15 @@
...
+@Name("All Afk Players")
 @Description("The list of afk players on the server.")
 @Examples({"send \"%all of the afk players%\""})
 @Since("1.0.0")
 public class ExprAfkPlayers extends SimpleExpression<Player> {
 
     static {
-        Skript.registerExpression(ExprAfkPlayers.class, Player.class, ExpressionType.SIMPLE, "[all [[of] the]|the] (afk|away from keyboard|idle) players");
+        Skript.registerExpression(ExprAfkPlayers.class, Player.class, ExpressionType.SIMPLE, "[all [[of] the]|the] (afk|away from keyboard|idle) players",
+                "players (in|with) (afk|away from keyboard|idle) (status|state|mode)");
```
Comment (sovdeeth) [inline src/main/java/site/equipable/skEssentials/skript/expressions/ExprAfkPlayers.java:33]: missing all the

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java [hunk 8]
```diff
@@ -28,7 +26,8 @@
...
+        Skript.registerExpression(ExprVanishedPlayers.class, Player.class, ExpressionType.SIMPLE, "[all [[of] the]|the] vanish[ed] players",
+                "players (in|with) vanish mode");
```
Comment (sovdeeth) [inline src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java:30]: missing all the

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java (hunk 8 above)
Comment (EquipableMC) [reply src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java:30]: no they are there?

Comment (EquipableMC) [reply src/main/java/site/equipable/skEssentials/skript/expressions/ExprVanishedPlayers.java:30]: Skript.registerExpression(ExprVanishedPlayers.class, Player.class, ExpressionType.SIMPLE, **"[all [[of] the]|the] vanish[ed] players"**

**Review by sovdeeth — APPROVED** [review]

**Review by cheeezburga — APPROVED** [review]

**Pull request merged**